modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionDependencyResolverImpl.java

```diff
@@ -81,8 +81,10 @@ public IgniteRel visit(IgniteTableScan rel) {
             private void resolveTable(IgniteTable igniteTable) {
                 int tableId = igniteTable.id();
                 TableDescriptor tableDescriptor = igniteTable.descriptor();
+                //TODO IGNITE-19499 Use id instead of name.
+                String tableName = igniteTable.name();
 
-                tableMap.computeIfAbsent(tableId, (id) -> registry.getTable(tableId, tableDescriptor));
+                tableMap.computeIfAbsent(tableId, (id) -> registry.getTable(schemaVersion, tableId, tableName, tableDescriptor));
```

Comment:
the only one place in production code where registry.getTable is called, looks a bit strange that we need both schemaVersion and tableDescriptor here.

Reply:
We need schemaVersion to get table from cache inside the `registry`.
TableDescriptor contains no id, and no schemaVersion.
Maybe we have to rethink, how tables are resolved.